onenote/endpoint.bal

```diff
@@ -0,0 +1,274 @@
+// Copyright (c) 2021 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+
+# Ballerina Microsoft OneNote connector provides the capability to access Microsoft Graph Excel API
+# It provides capability to perform perform CRUD (Create, Read, Update, and Delete) operations on [OneNote]
+# (https://docs.microsoft.com/en-us/graph/api/resources/onenote-api-overview?view=graph-rest-1.0).
+# 
+# + httpClient - HTTP Client
+@display {
+    label: "Microsoft OneNote Client"
+}
+public isolated client class Client {
+    private final http:Client httpClient;
+
+    # Initializes the connector. During initialization you can pass either http:BearerTokenConfig if you have a bearer 
+    # token or http:OAuth2RefreshTokenGrantConfig if you have OAuth tokens.
+    # Create a Microsoft account and obtain tokens following 
+    # [this guide](https://docs.microsoft.com/en-us/azure/active-directory/develop/active-directory-v2-protocols)
+    # 
+    # + config - Configuration required to initialize the client
+    # + return - An error if initialization fails
+    public isolated function init(Configuration config) returns error? {
+        self.httpClient = check new (BASE_URL, {
+            auth: config.authConfig,
+            secureSocket: config?.secureSocketConfig
+        });
+    }
+
+    // Notebooks
+
+    # Creates a notebook.
+    # 
+    # + noteBookName - Display name for notebook
+    # + return - Notebook if successful or else an error
+    @display {label: "Create Notebook"}
+    remote isolated function createNotebook(@display {label: "Notebook Name"} string noteBookName) returns 
+                                            Notebook|error {
+        string path = NOTEBOOKS_PATH;
+        http:Response response = check self.httpClient->post(path, {"displayName": noteBookName});
+        map<json> handledResponse = check handleResponse(response);
+        return check handledResponse.cloneWithType(Notebook);
```

Comment:
This will work fine in the successful scenario.

But i have implemented a function called handleResponse and checking the statusCode and handling the mapping manually. This is better comparing to the usage of targetType when we have errors.

Reply:
Did the change according to the suggestion
(targetType)